File: 2998-count-symmetric-integers/2998-count-symmetric-integers.ts
function countSymmetricIntegers(low: number, high: number): number {
    let cnt = 0;
    for (let i = low; i <= high; i++) {
        const strg = i.toString();
        if (strg.length % 2) continue;
        const sum = strg.split('').reduce((acc, curr) => acc + parseInt(curr), 0);
        let half_sum = 0;
        for (let i = 0; i < strg.length / 2; i++) {
            half_sum += parseInt(strg[i]);
        }
        if (2 * half_sum === sum) cnt++;
    }

    return cnt;
};

File: 2998-count-symmetric-integers/2998-count-symmetric-integers.java
class Solution {
    public int countSymmetricIntegers(int low, int high) {
        int cnt = 0;

        for (int i = low; i <= high; i++) {
            String str = Integer.toString(i);
            int len = str.length();
            if (len % 2 != 0)
                continue;

            int leftSum = 0;
            int rightSum = 0;
            int mid = len / 2;

            for (int j = 0; j < mid; j++) {
                leftSum += str.charAt(j) - '0';
                rightSum += str.charAt(mid + j) - '0';
            }

            if (leftSum == rightSum)
                cnt++;
        }

        return cnt;

    }
}